src/type.zig

```diff
@@ -1805,7 +1805,13 @@ pub const Type = extern union {
             .void,
             => 0,
 
-            .@"struct" => return self.structFieldOffset(self.structFieldCount(), target),
+            .@"struct" => {
+                const field_count = self.structFieldCount();
+                if (field_count == 0) {
+                    return 0;
+                }
+                return self.structFieldOffset(self.structFieldCount(), target);
```

Comment:
Looks like you meant to do this: 

```suggestion
                return self.structFieldOffset(field_count, target);
```